File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.cpp
class Solution {
public:
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> t;
        int res = 0;
        for(auto x: tasks) {
            if(t.find(x) == t.end())
                t[x] = 1;
            else
                t[x] += 1;
        }
        for(auto x: t) {
            if(x.second == 1)
                return -1;
            else
                res += x.second / 3 + (x.second - 3 * floor(x.second / 3) == 1 || x.second - 3 * floor(x.second / 3) == 2);
        }
        return res;
    }
};

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        c = dict(Counter(tasks))
        print(c)
        res = 0
        for value in c.values():
            if value == 1:
                return -1
            else:
                res += value // 3 + ((value - 3 * int(value / 3)) == 1 or value - 3 * int(value / 3) == 2)
        return res